### PR TITLE
better error message in discretize.

### PR DESCRIPTION
**Pull request by weinbe58**

### CODE REVIEW

**Review by Wang-Shengtao — CHANGES_REQUESTED** [review]
replace all `Descretization` to `Discretization`. Most probably at other places as well. Otherwise, looks good to me.

File: lib/BloqadeWaveforms/src/discretize.jl [hunk 1]
```diff
@@ -63,8 +63,10 @@ function discretize(wf::Waveform{PiecewiseConstant{T},T};
         Δt = 4*itol/abs(Δv) # let error determine step size, error is equal to area between step and line functions
         slope = Δv/Δt
         
-        if Δt < min_step || abs(slope) > max_slope
-            throw(ErrorException("Descretization cannot obtain requested tolerance given the step size and slope constraints."))
+        if abs(slope) > max_slope
+            throw(ErrorException("Descretization cannot obtain requested tolerance given the slope constraint."))
```
Comment (weinbe58) [inline lib/BloqadeWaveforms/src/discretize.jl:67]:
```suggestion
            throw(ErrorException("Discretization cannot obtain requested tolerance given the slope constraint."))
```

File: lib/BloqadeWaveforms/src/discretize.jl [hunk 2]
```diff
@@ -63,8 +63,10 @@ function discretize(wf::Waveform{PiecewiseConstant{T},T};
         Δt = 4*itol/abs(Δv) # let error determine step size, error is equal to area between step and line functions
         slope = Δv/Δt
         
-        if Δt < min_step || abs(slope) > max_slope
-            throw(ErrorException("Descretization cannot obtain requested tolerance given the step size and slope constraints."))
+        if abs(slope) > max_slope
+            throw(ErrorException("Discretization cannot obtain requested tolerance given the slope constraint."))
+        elseif Δt < min_step
+            throw(ErrorException("Descretization cannot obtain requested tolerance given the step size constraint."))
```
Comment (weinbe58) [inline lib/BloqadeWaveforms/src/discretize.jl:69]:
```suggestion
            throw(ErrorException("Discretization cannot obtain requested tolerance given the step size constraint."))
```

File: lib/BloqadeWaveforms/src/discretize.jl [hunk 3]
```diff
@@ -137,9 +139,11 @@ function discretize(wf::Waveform;
 
             next_slope = 2*max(abs(f_mid - f_lb),abs(f_ub - f_mid))/interval
             
-            if interval < 2*min_step || next_slope > max_slope
-                # can't satisfy constraint. stop refining grid
-                throw(ErrorException("Descretization cannot obtain requested tolerance given the step size and slope constraints."))
+
+            if next_slope > max_slope
+                throw(ErrorException("Descretization cannot obtain requested tolerance given the slope constraint."))
```
Comment (weinbe58) [inline lib/BloqadeWaveforms/src/discretize.jl:144]:
```suggestion
                throw(ErrorException("Discretization cannot obtain requested tolerance given the slope constraint."))
```

File: lib/BloqadeWaveforms/src/discretize.jl [hunk 4]
```diff
@@ -137,9 +139,11 @@ function discretize(wf::Waveform;
...
+            elseif interval < 2*min_step
+                    throw(ErrorException("Descretization cannot obtain requested tolerance given the step size constraint."))
```
Comment (weinbe58) [inline lib/BloqadeWaveforms/src/discretize.jl:146]:
```suggestion
                    throw(ErrorException("Discretization cannot obtain requested tolerance given the step size constraint."))
```

**Review by weinbe58 — COMMENTED** [review]
Should be all set.

**Review by Wang-Shengtao — APPROVED** [review]

**Pull request merged**